### DISCUSSION TITLE
Crespo Hernandez model constants

### DISCUSSION THREAD

**Discussion by prith-gs** [NatLabRockies/floris#995] [General]
Hello,

Why are the constants for the crespo hernandez model given in the examples different from the original paper?

Here's the equation from the original paper
![image](https://github.com/user-attachments/assets/8921a9de-453c-4775-b85c-c0067437111e)

![image](https://github.com/user-attachments/assets/fa4641f7-5c9a-4fba-abc9-cb5bfb47bb52)

Can someone explain why the constants from the original paper were not used? Thanks!

**Comment by Zhiming-Cai** [comment]
Hello! I have the same question as you.
I checked the same original paper you did, I also noticed that the parameters defined in the crespo_hernandez.py file in floris-v4 are different from those defined in the original paper.
![image](https://github.com/user-attachments/assets/38e481dc-937e-418d-9e58-0502d8648e45)
As shown in the figure above, 'initial' is defined as the initial ambient turbulence intensity.👆
![image](https://github.com/user-attachments/assets/1066e8c6-ca6f-48d0-99df-feca090b8852)
However, in formula calculation, 'initial' appears as the exponent of ambient_TI.👆
All in all, the parameters definition here is very questionable.🤨

**Comment by Zhiming-Cai** [comment]
> Hello,
> 
> Why are the constants for the crespo hernandez model given in the examples different from the original paper?
> 
> Here's the equation from the original paper ![image](https://private-user-images.githubusercontent.com/103917694/374554638-8921a9de-453c-4775-b85c-c0067437111e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxNjQxMDYsIm5iZiI6MTcyOTE2MzgwNiwicGF0aCI6Ii8xMDM5MTc2OTQvMzc0NTU0NjM4LTg5MjFhOWRlLTQ1M2MtNDc3NS1iODVjLWMwMDY3NDM3MTExZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxN1QxMTE2NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MDgxNGJhN2U5OWYxNDFiZTQzN2FiNzM0MGI3MTJiYmZjNjlkNDk1MTUwZmJlOGZlMjA1Mjg1NDBhMjQzYzU1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4HbumKp1fFTq0EYKqm9uCOCEAd5IIC1fjwSWnFCOdhY)
> 
> ![image](https://private-user-images.githubusercontent.com/103917694/374554738-fa4641f7-5c9a-4fba-abc9-cb5bfb47bb52.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxNjQxMDYsIm5iZiI6MTcyOTE2MzgwNiwicGF0aCI6Ii8xMDM5MTc2OTQvMzc0NTU0NzM4LWZhNDY0MWY3LTVjOWEtNGZiYS1hYmM5LWNiNWJmYjQ3YmI1Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxN1QxMTE2NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZDEyODVmYjRhMzdmOTFkZjRhMTVkMWVlMDhlNTVhN2IzZDIxMTRhYzI4OWZkYjM0ZTIzNjY0ZmY1YWZiZjQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.g7rRXZQTAnVTtYlS4P3i1zy-kv4alikQ4VZ05L2GLRA)
> 
> Can someone explain why the constants from the original paper were not used? Thanks!

Hi. I found the source of the NREL data.
Crespo, A., Hernández, J., and Frandsen, S.: Survey of modelling methods for wind turbine wakes and wind farms, Wind Energy, 2, 1–24, 1999.👇
![image](https://github.com/user-attachments/assets/f09dc300-94c9-4044-9776-6ed4b04ea83d)
But I have not verified whether the data is actually from this article.

**Comment by misi9170** [comment]
Hi @prith-gs , @CaiZhiming-HHU ,

@CaiZhiming-HHU , thank you for answering this! To be clear, I believe the default values in FLORIS (and stated in [King et al. (2021)](https://wes.copernicus.org/articles/6/701/2021/)) are not a direct match for previous published results. Rather, as stated in the paragraph following eq. (5) of King et al. (2021), they use the same form, but parameters have been retuned to better agree with LES:

> The values used in this equation are slightly different from those in Niayifar and Porté-Agel (2015) and have been tuned to large-eddy simulations.

It is also interesting to note that [Niayifar and Porté-Agel (2015)](https://iopscience.iop.org/article/10.1088/1742-6596/625/1/012039) apparently use slightly different values to those in [Crespo and Hernández (1996)](https://www.sciencedirect.com/science/article/pii/016761059500033X), with the exponent on $I_\infty$ changing from $-0.0325$ in Crespo and Hernández (1996) to $0.0325$ in Niayifar and Porté-Agel (2015). This could possibly be from a PDF rendering issue in the Crespo and Hernández paper---in my PDF viewer, the minus sign is very faint (and I probably would not have noticed it at all if @prith-gs hadn't reproduced it above with the minus sign very clear). For a value of $I_\infty = 0.06$, as an example, the difference between these is not huge, but is noticeable ($I_\infty^{-0.0325} = 1.096$ whereas $I_\infty^{0.0325} = 0.913$).

Not to state the obvious, but you are of course very free to change the parameter values to the original ones reported by Crespo and Hernández (1996)!
